test/functional/normalization/audit_logs_format_test.go

```diff
@@ -0,0 +1,249 @@
+			// Log message data
+			timestamp := "2013-03-28T14:36:03.243000+00:00"
+			testTime, _ := time.Parse(time.RFC3339Nano, timestamp)
+			auditLogLine := functional.NewAuditHostLog(testTime)
+			// Template expected as output Log
+			var outputLogTemplate = types.LinuxAuditLog{
+				Message:  auditLogLine,
+				LogType:  "audit",
+				Hostname: functional.FunctionalNodeName,
+				AuditLinux: types.AuditLinux{
+					Type:     "DAEMON_START",
+					RecordID: "*",
+				},
+				Timestamp:        testTime,
+				ViaqMsgID:        "*",
+				PipelineMetadata: functional.TemplateForAnyPipelineMetadata,
+			}
+			// Write log line as input to fluentd
+			Expect(framework.WriteMessagesToAuditLog(auditLogLine, 10)).To(BeNil())
+			// Read line from Log Forward output
+			raw, err := framework.ReadAuditLogsFrom(logging.OutputTypeFluentdForward)
+			Expect(err).To(BeNil(), "Expected no errors reading the logs")
+			var logs []types.LinuxAuditLog
+			err = types.StrictlyParseLogs(utils.ToJsonLogs(raw), &logs)
+			Expect(err).To(BeNil(), "Expected no errors parsing the logs")
+			// Compare to expected template
+			outputTestLog := logs[0]
+			Expect(outputTestLog).To(FitLogFormatTemplate(outputLogTemplate))
+		})
+
+		It("should parse ovn audit log correctly", func() {
+			// Log message data
+			level := "info"
+			ovnLogLine := functional.NewOVNAuditLog(time.Now())
+
+			// Template expected as output Log
+			var outputLogTemplate = types.OVNAuditLog{
+				Message:   ovnLogLine,
+				Level:     level,
+				Hostname:  functional.FunctionalNodeName,
+				Timestamp: time.Time{},
+				LogType:   "audit",
+				Openshift: types.OpenshiftMeta{
+					Sequence: types.NewOptionalInt(""),
+				},
+				ViaqMsgID:        "*",
+				PipelineMetadata: functional.TemplateForAnyPipelineMetadata,
+			}
+			outputLogTemplate.PipelineMetadata.Collector.ReceivedAt = time.Time{}
+			// Write log line as input to fluentd
+			Expect(framework.WriteMessagesToOVNAuditLog(ovnLogLine, 10)).To(BeNil())
+			// Read line from Log Forward output
+			raw, err := framework.ReadOvnAuditLogsFrom(logging.OutputTypeFluentdForward)
+			Expect(err).To(BeNil(), "Expected no errors reading the logs")
+			var logs []types.OVNAuditLog
+			err = types.StrictlyParseLogs(utils.ToJsonLogs(raw), &logs)
+			ExpectOK(err)
+			// Compare to expected template
+			outputTestLog := logs[0]
+			Expect(outputTestLog).To(FitLogFormatTemplate(outputLogTemplate))
+		})
+
+		AfterEach(func() {
+			framework.Cleanup()
+		})
+	})
+
+	Context("with vector", func() {
+		BeforeEach(func() {
+			framework = functional.NewCollectorFunctionalFrameworkUsingCollector(logging.LogCollectionTypeVector)
+			functional.NewClusterLogForwarderBuilder(framework.Forwarder).
+				FromInput(logging.InputNameAudit).
+				ToElasticSearchOutput()
+			Expect(framework.Deploy()).To(BeNil())
+		})
+
+		//TODO: fix me when audit formatting is enabled
+		It("should parse k8s audit log format correctly", func() {
```

Comment:
after the `TODO` is completed, can we use same `It` clause for both vector and fluentd?

Reply:
@vimalk78 I believe the answer to be "yes" but as you noted it requires us to change the destination because vector does not support Fluent

Reply:
yes